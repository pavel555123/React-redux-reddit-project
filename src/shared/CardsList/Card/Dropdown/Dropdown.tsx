import React, {useEffect, useState} from 'react';
import styles from './dropdown.css';

interface IDropdownProps {
  button: React.ReactNode;
  children: React.ReactNode;
  isOpen?: boolean;
  onOpen?: () => void;
  onClose?: () => void;
}

const NOOP = () => {
};

export function Dropdown({button, children, isOpen, onOpen = NOOP, onClose = NOOP}: IDropdownProps) {
  const [isDropdownOpen, setIsDropdownOpen] = useState(isOpen);
  useEffect(() => {
    setIsDropdownOpen(isOpen);
  }, [isOpen])
  useEffect(() => {
    isDropdownOpen ? onOpen() : onClose();
  }, [isDropdownOpen])

  const handleOpen = () => {
    if (isOpen === undefined) {
      setIsDropdownOpen(!isDropdownOpen);
    }
  }

  return (
      <div className={styles.container}>
        <div onClick={() => handleOpen()}>
          {button}
        </div>
        {isDropdownOpen && (
            <div className={styles.listContainer}>
              <div className={styles.list} onClick={() => setIsDropdownOpen(false)}>
                {children}
              </div>
            </div>
        )}
      </div>
  );
}

