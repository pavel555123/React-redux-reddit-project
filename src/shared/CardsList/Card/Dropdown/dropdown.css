.container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;
}

.listContainer {
    position: relative;
}

.list {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
}