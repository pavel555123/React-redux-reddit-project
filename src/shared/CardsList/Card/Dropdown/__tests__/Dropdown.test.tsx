import React from "react";
import {Dropdown} from "../Dropdown";
import {configure, shallow} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
describe('Dropdown', () => {
    test('should render', () => {
        const wrapper = shallow(<Dropdown children={<div>777</div>} button={<button>555</button>}/>)
        expect(wrapper).toBeDefined();
        expect(wrapper.find('div.container').isEmptyRender()).toBeFalsy();
    })

    test('should render (snapshot', () => {
        const wrapper = shallow(<Dropdown children={<div>777</div>} button={<button>555</button>}/>)

        expect(wrapper).toMatchSnapshot();
    })
});