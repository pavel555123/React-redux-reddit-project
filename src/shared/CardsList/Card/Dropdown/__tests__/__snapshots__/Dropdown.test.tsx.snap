// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render (snapshot 1`] = `
<div
  className="container"
>
  <div
    onClick={[Function]}
  >
    <button>
      555
    </button>
  </div>
</div>
`;
